catalog/app/components/Form/TextField.tsx

```diff
@@ -11,20 +11,26 @@ const useFieldInputStyles = M.makeStyles({
 
 interface FieldProps {
   error?: string
+  errors: Record<string, React.ReactNode>
   helperText?: React.ReactNode
   input: RF.FieldInputProps<string>
+  meta: RF.FieldMetaState<string>
   validating?: boolean
 }
 
 export default function Field({
   InputLabelProps,
-  error,
   helperText,
   input,
   validating,
+  errors,
+  meta,
   ...rest
 }: FieldProps & M.TextFieldProps) {
   const inputClasses = useFieldInputStyles()
+  const readyForValidation = (input.value && meta.modified) || meta.submitFailed
```

Comment:
does this mean the field would be ready for validation after a single letter is entered? doesn't seem like a good UX

Reply:
I think there are some single letter names

Reply:
final-form validates anyway
This check is just about showing validation error. So, better naming would be something like `readyToShowValidationError`

Reply:
i meant that we probably shouldn't show validation errors until the form is submitted (we do this everywhere across the app), bc it's quite bad UX to show the errors right away and especially as you type (it feels quite annoying)

Reply:
> we do this everywhere across the app
Except one place from where I copied that piece of code :)

Fixed, thanks